src/t_hash.c

```diff
@@ -1697,15 +1737,19 @@ void hpersistCommand(client *c) {
     if (checkType(c, hash, OBJ_HASH))
         return;
 
+    /* Remember current volatile state to detect changes after modification. */
+    bool has_vola = hash && hashTypeHasVolatileElements(hash);
+
     for (int i = 0; i < num_fields; i++, fields_index++) {
-        result = hashTypePersist(hash, c->argv[fields_index]->ptr);
+        result = hashTypePersist(c->db, hash, c->argv[fields_index]->ptr);
         if (result > 0) {
             server.dirty++;
             changes++;
         }
         addReplyLongLong(c, result);
     }
     if (changes) {
+        updateVolatileTrackingIfNeeded(c->db, hash, has_vola);
```

Comment:
is this needed? hashTypePersist should untrack and remove from the kvstore right? what am I missing?